### PR TITLE
API.md: rename "domain" to "tailnet" in documentation

### PR DESCRIPTION
**Pull request by c22wen**
Added the following:
* A brief explanation of "domain".
* A brief explanations of "device" and how to find a "deviceID".
* Renamed "domain" to "tailnet"
* Sample API format for solo plans.
* `DELETE /device/:deviceID` documentation

### CODE REVIEW

**Review by rosszurowski — APPROVED** [review]
I wasn't marked as a reviewer, but this description is concise and makes sense to me. Props for also linking our "inviting others to your network" docs.

**Comment by bradfitz** [comment]
I actually think we want to remove "domain" from the API docs entirely and replace it with "tailnet" or "network" instead.

We can keep any "domain" named fields still working for old users but we shouldn't document it.

**Review by chungdaniel — APPROVED** [review]

File: api.md [hunk 1]
```diff
@@ -97,6 +103,43 @@ Response
 }
 ```
 
+
+#### `DELETE /api/v2/device/:deviceID` - deletes the device from your tailnet
```
Comment (bradfitz) [inline api.md:107]: From which tailnet?

Or is an API key bound to a specific tailnet that's implicit? But if so, why do the other APIs require a tailnet ID?

File: api.md [hunk 2]
```diff
@@ -97,6 +103,43 @@ Response
...
+Deletes the provided device from your tailnet. 
+The device must belong to your tailnet.
```
Comment (bradfitz) [inline api.md:109]: "your tailnet" again implies you can only have one, which is currently true but not something I believe we want to assume going forward

**Review by crawshaw — APPROVED** [review]

**Pull request merged**